pkg/util/secretcode/totp.go

```diff
@@ -13,6 +13,16 @@ import (
 	"github.com/pquerna/otp/totp"
 )
 
+var TOTPSecretCode = TOTPSecretCodeType{}
+
+type TOTPSecretCodeType struct{}
+
+type MakeTOTPKeyOptions struct {
+	Issuer      string
+	AccountName string
+	Secret      string
+}
+
```

Comment:
- Remove `type ValidateOpts`
- Make `ValidateOptsTOTP` private
- Redesign the API of TOTP

```golang
type TOTP struct {
  Secret string
}

type QRCodeImageOptions struct {
  Issuer string
  AccountName string
  Widget int
  Height int
}

func NewTOTPFromRNG() (*TOTP, error)
func NewTOTPFromSecret(secret string) *TOTP
func (c *TOTP) GenerateCode(t time.Time) (string, error)
func (c *TOTP) ValidateCode(t time.Time, code string) bool
func (c *TOTP) QRCodeImage(opts QRCodeImageOptions) (image.Image, error)

func TOTPCheckFormat(value interface{}) error
```

The rationale of the redesign is that TOTPSecretCodeType is not a very useful type. It contains no data. Its methods can simply be turned into ordinary functions.

But TOTP is different. It contains the secret. It encapsulates the secret and exposes 3 methods for us to interact with it. The caller cannot misuse the concept of `secret` and `code` here. To use TOTP, we have to choose from NewTOTPFromRNG or NewTOTPFromSecret. From the design of the API, the caller can learn that TOTP contains secret. GenerateCode and ValidateCode advertise that TOTP is related to time, as both methods take a time argument.